providers/implementations/keymgmt/dh_kmgmt.c

```diff
@@ -814,6 +814,16 @@ static void *dh_gen(void *genctx, OSSL_CALLBACK *osslcb, void *cbarg)
     DH_clear_flags(dh, DH_FLAG_TYPE_MASK);
     DH_set_flags(dh, gctx->dh_type);
 
+#ifdef FIPS_MODULE
+    if (!ossl_fips_self_testing()) {
```

Comment:
Shouldn't this whole block be inside the `gctx->selection & OSSL_KEYMGMT_SELECT_KEYPAIR) != 0` section? If we're doing a PCT then presumably it only makes sense if we generated a key pair (as opposed to just generating parameters).

Reply:
This is most likely the cause of the CI failure. Just move this all inside the block ending at line 813.